src/jit-format/jit-format.cs

```diff
@@ -0,0 +1,682 @@
+                }
+
+                SetPlatform();
+
+                if (_verbose)
+                {
+                    Console.WriteLine("Operating system is {0}", _os);
+                }
+            }
+
+            if (_os == "Windows")
+            {
+                _os = "Windows_NT";
+            }
+
+            if (_srcDirectory == null)
+            {
+                if (_verbose)
+                {
+                    Console.WriteLine("Formatting jit directory.");
+                }
+                _srcDirectory = "jit";
+            }
+
+            if (_projects.Count == 0 && _verbose)
+            {
+                Console.WriteLine("Formatting dll project.");
+            }
+
+            if (!_untidy && ( (_arch == null) || (_os == null) || (_build == null)))
+            {
+                _syntaxResult.ReportError("Specify --arch, --plaform, and --build for clang-tidy run.");
+            }
+
+            if (_rootPath == null)
+            {
+                _syntaxResult.ReportError("Specify --coreclr");
+            }
+
+            // Check that we can find the build log for Windows.
+            if (_os == "Windows_NT")
+            {
+                string logFile = "CoreCLR_Windows_NT__" + _arch + "__" + _build + ".log";
+                string logFullPath = Path.Combine(_rootPath, "bin", "Logs", logFile);
+                if (_compileCommands != null)
+                {
+                    _buildCompileCommands = false;
+                }
+                else if (!_untidy && !File.Exists(logFullPath))
+                {
+                    _syntaxResult.ReportError("Can't find build log.");
+                }
+                else
+                {
+                    _buildLog = logFullPath;
+                    string[] compileCommandsPath = { _rootPath, "bin", "obj", "Windows_NT." + _arch + "." + _build, "compile_commands.json" };
+                    _compileCommands = Path.Combine(compileCommandsPath);
+                }
+            }
+
+            // Check that we can find the compile_commands.json file on other platforms
+            else
+            {
+                string[] compileCommandsPath = { _rootPath, "bin", "obj", _os + "." + _arch + "." + _build, "compile_commands.json" };
+                if (!_untidy && !File.Exists(Path.Combine(compileCommandsPath)))
+                {
+                    _syntaxResult.ReportError("Can't find compile_commands.json file. Please build coreclr first.");
+                }
+                else
+                {
+                    _compileCommands = Path.Combine(compileCommandsPath);
+                }
+            }
+        }
+
+        private void LoadFileConfig()
+        {
+            _jitUtilsRoot = Environment.GetEnvironmentVariable("JIT_UTILS_ROOT");
+
+            if (_jitUtilsRoot != null)
+            {
+                string path = Path.Combine(_jitUtilsRoot, "format.json");
```

Comment:
What sorts of additional changes would need to be made to the json parsing to make that work? I don't really know much about this, so I just went with duplicating the logic with its own file, since I knew I could make that work.